frontend/src/components/Dropdown/index.ts

```diff
@@ -0,0 +1,77 @@
+import { createElement } from "@/utils/dom";
+import "./style.scss";
+
+export interface IDropdownItem {
+  text: string;
+  danger?: boolean;
+  className?: string;
+  id?: string;
+  onClick?: (event: any) => void;
+}
+
+export interface IDropdownProps {
+  style: {
+    [key: string]: string;
+  };
+  items: IDropdownItem[];
+  delegate?: (event: any) => void;
+  parentSelector: string;
+}
+
+const template = (props: IDropdownProps): string => `
+  <div class="dropdown" style="${makeStyle(props.style)}">
+    ${props.items.map((item: IDropdownItem) => makeDropdownItem(item)).join("")}
+  </div>
+`;
+
+export default function Dropdown(props: IDropdownProps): HTMLElement {
+  const { items, delegate, parentSelector } = props;
+  const $dropdown = createElement(template(props));
+  const $dropdownItems = $dropdown.children;
+
+  if (delegate) {
+    $dropdown.addEventListener("click", delegate);
+  }
+  items.forEach((item, i) => {
+    const { onClick } = item;
+    if (onClick) {
+      $dropdownItems[i].addEventListener("click", onClick);
+    }
+  });
+
+  document.body.addEventListener(
+    "click",
+    removeDropdown($dropdown, parentSelector)
+  );
+
+  return $dropdown;
+}
+
+function makeDropdownItem(item: IDropdownItem) {
+  const { text, danger, className, id } = item;
+  let selector = "";
+  if (id) selector += `id=${id}`;
+  if (danger) {
+    return `<div class="dropdown__item dropdown__item--danger ${
+      className || ""
+    }" ${selector}>${text}</div>`;
+  }
```

Comment:
```suggestion
  if (danger) {
  return `
    <div 
      ${id ? `id=${id}` : ""}
      class="dropdown__item ${className || ""}" 
    >
      ${text}
    </div>
  `;
  }
```
selector 부분은 이런식으로도 표현 가능할 것 같아요.

Reply:
삼항연산자로 바꿔보겠습니다 !